### ISSUE TITLE
Installing Windows Terminal changes the behavior of Start menu

### ISSUE THREAD

**Issue by oruponu** [microsoft/terminal#7122]
# Environment

```none
Windows build number: 10.0.19041.0
Windows Terminal version (if applicable): 1.1.2021.0
```

# Steps to reproduce

1. Open Start menu.
2. Click on pinned folder tile or click on folder icon in the bottom left corner (Documents, Pictures, etc.) to open it.
Settings icon in the bottom left corner and Application tiles are not affected.

# Expected behavior

Start menu will automatically close and Explorer will open.

# Actual behavior

Start menu will remain open and Explorer will open.

***
I uninstall Windows Terminal and confirmed that Start menu works as expected.
I also confirmed that the same issue occurs on other computers.

**Comment by zadjii-msft** [comment]
Well that's totally crazy. I don't have the faintest idea why that might happen. I also can't repro this locally, so I can't really investigate this further. Did this happen with previous version of the Terminal? Or only after upgrading to 1.1?

**Comment by fran-f** [comment]
I can see this on two different machine. I think both are running Windows 10.0.18363.959.

The issue appears with Terminal 1.1, installed from store or package, but not with 1.0 (at least installed with package). It presents itself immediately after installation, no need to launch terminal once.

(Unfortunately I also discovered that if you uninstall Terminal your configuration is deleted...)

**Comment by DHowett** [comment]
This is wild. I can reproduce this! As stated, it only repros with 1.1+, and I have the sneaking suspicion that it's related to the shell extension.

Seems like a platform issue. I'll see if I can set up a minimal repro.

**Comment by fran-f** [comment]
Is the shell extension enabled by default? I don't see an entry in the context menu.

**Comment by DHowett** [comment]
@fran-f yes, but due to a Windows issue it's not available if you right-click in the empty space in a directory window. Until that bug is fixed, it's only available if you actually right-click _on a directory_.

**Comment by fran-f** [comment]
I see! Then it's active, I can see the entry right-clicking on directories. However I can't find it in [ShellExView](http://www.nirsoft.net/utils/shexview.html), so I can't try to disable it.

**Comment by DHowett** [comment]
Yeah, packaged shell extensions are (we're learning!) very unusual. Somebody suggested a workaround (to disable it) here: https://github.com/microsoft/terminal/issues/7008#issuecomment-662621638. You'll almost certainly need to restart Explorer.

**Comment by fran-f** [comment]
Following that workaround caused a curious restart loop in Explorer, but after a logout the extension is disabled, and the issue with the start menu is gone.

**Comment by DHowett** [comment]
Thanks for confirming!

**Comment by oruponu** [comment]
Thank you.
I have also confirmed that the Start menu works as expected with that workaround.

**Comment by fran-f** [comment]
As a shortcut to enable or disable the shell extension, one can use a `.reg` file. I can't upload them to GitHub, but I'll dump them here if anyone needs them:

`disable-windows-terminal-shell-extension.reg`
```
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Shell Extensions\Blocked]
"{9F156763-7844-4DC4-B2B1-901F640F5155}"="WindowsTerminal"
```

`re-enable-windows-terminal-shell-extension.reg`
```
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Shell Extensions\Blocked]
"{9F156763-7844-4DC4-B2B1-901F640F5155}"=-
```

**Comment by zadjii-msft** [comment]
Just linking this to the internal thread: MSFT:32962688

**Comment by Ronald-G-Andrade** [comment]
Hello dear friends, just a reminder that this bug is still present in the latest stable version (v1.10.2383.0).
Click to open a pinned folder in the start menu (also in full-screen mode) won't close it.
Thank you very much 👍🏻

**Comment by zadjii-msft** [comment]
Yep. There's nothing the Terminal can do about this, it's a Start Menu bug. We'll make sure to update this thread when there is anything to share. The internal bug is still open, but this also doesn't repro on Windows 11 anymore, so I'd bet that this isn't something that ends up getting serviced back to Windows 10.

**Comment by stjpInd** [comment]
I haven't done any extensive testing, but it looks like the issue might be fixed now!
I'm currently on Windows 10 build 19044.1387 with Terminal 1.11.2921.0 installed (no registry workaround) and I don't see the broken Start menu behavior anymore. Pinned folders and search both work as expected.
My suspicion is that Microsoft may have fixed the bug around the same time they rolled out the new search with dark mode, but I couldn't say for certain.

**Comment by Ronald-G-Andrade** [comment]
> I haven't done any extensive testing, but it looks like the issue might be fixed now! I'm currently on Windows 10 build 19044.1387 with Terminal 1.11.2921.0 installed (no registry workaround) and I don't see the broken Start menu behavior anymore. Pinned folders and search both work as expected. My suspicion is that Microsoft may have fixed the bug around the same time they rolled out the new search with dark mode, but I couldn't say for certain.

It's true dear brother. I tested it and now it's working properly on Windows, either an app or folder, it opens normally by closing the start menu / start screen.

Now, I really hope that Microsoft won't mess up again, later we would fix it and say: ""A new feature"" jajaja

**Comment by zadjii-msft** [comment]
Alrighty folks, looks like the explorer team looked into this and found that it doesn't repro on Windows 11, so it must have been fixed as part of the Start Menu rejuvenation. Impossible to say which changelist as a part of that would have fixed this, so it's not gonna be possible to service that fix to Windows 10. Sorry about that. 

If folks are seeing this on Windows 11 builds, ping us here on this thread and I can reopen the investigation with the explorer folks. 

Thanks!

**Comment by pgorod** [comment]
> Microsoft will continue to support at least one Windows 10 release until October 14, 2025.

... plus the usual years-long delays.

Is this really proper to just ditch Windows 10 bugs as "won't fix" at the current time? Isn't it too early for that? 

This saddens me, this bug is really annoying, and it's not even something that only happens inside Windows Terminal, it affects the general OS experience. I am enjoying Terminal but this is making me seriously consider uninstalling it. Please reconsider.

**Comment by DHowett** [comment]
> Please reconsider.

Unfortunately, it's just not our call. We only own the Terminal, the console subsystem, and the console driver... not parts of explorer or the start menu or the taskbar. We only have so much power over our peers. :smile:

**Comment by merrickjeff** [comment]
> > Microsoft will continue to support at least one Windows 10 release until October 14, 2025.
> 
> ... plus the usual years-long delays.
> 
> Is this really proper to just ditch Windows 10 bugs as "won't fix" at the current time? Isn't it too early for that?
> 
> This saddens me, this bug is really annoying, and it's not even something that only happens inside Windows Terminal, it affects the general OS experience. I am enjoying Terminal but this is making me seriously consider uninstalling it. Please reconsider.

@pgorod Either of these fixes should work unless you really need that context menu: 
https://github.com/microsoft/terminal/issues/7008#issuecomment-662621638
https://github.com/microsoft/terminal/issues/7122#issuecomment-683299285

**Comment by hello-smile6** [comment]
> > > Microsoft will continue to support at least one Windows 10 release until October 14, 2025.
> > 
> > 
> > ... plus the usual years-long delays.
> > Is this really proper to just ditch Windows 10 bugs as "won't fix" at the current time? Isn't it too early for that?
> > This saddens me, this bug is really annoying, and it's not even something that only happens inside Windows Terminal, it affects the general OS experience. I am enjoying Terminal but this is making me seriously consider uninstalling it. Please reconsider.
> 
> @pgorod Either of these fixes should work unless you really need that context menu: [#7008 (comment)](https://github.com/microsoft/terminal/issues/7008#issuecomment-662621638) [#7122 (comment)](https://github.com/microsoft/terminal/issues/7122#issuecomment-683299285)

Any workarounds that don't require TPM 2.0?

**Comment by DHowett** [comment]
> Any workarounds that don't require TPM 2.0?

You literally quoted a message that called out two workarounds that don't require _installing Windows 11._

**Comment by DHowett** [comment]
> at least you can try telling them "people are complaining that there are still years of support for Windows 10 to exhaust before you get a right to start dropping bugs"

Sure can try!

**Comment by hello-smile6** [comment]
> > Any workarounds that don't require TPM 2.0?
> 
> You literally quoted a message that called out two workarounds that don't require _installing Windows 11._

Oh

**Comment by dymanoid** [comment]
Unfortunately, the suggested workarounds described above (disabling the wt shell extension) didn't work for me.
I can confirm there's no "Open in Windows Terminal" context menu item anymore, but the Start menu behavior is still broken.
I recently installed Visual Studio 2022 Community though, so there might be some other shell extensions causing this behavior. (It might be a coincidence, but I don't observe this behavior on another Windows 10 machine having Windows Terminal but no Visual Studio 2022 installed.)

**Comment by mattsmith-cpi** [comment]
I didn't even know there _was_ a "open Terminal here" extension. I don't even use Windows Terminal - I just had it installed passively -- my habit is still to press the Windows key, type the letters "cmd", and hit Enter or Ctrl+Shift+Enter. Which, on random computers, for months now, had occasionally resulted in cmd (or any other application I open from the start menu) opening up, behind the start menu, leaving me to additionally close the start menu -- or worse, start interacting with the start menu with the keyboard instead of the application (I work very quickly sometimes - this usually happens in the span of 1 second or so).

So, Terminal was identified as the reason for this, eh? And the Terminal folks are like, "😊 haha well it's a Windows bug, we can't fix it"?

I had to become frustrated enough with this issue to even find a way to phrase it to search it - and then trust the absurd conclusion that _an app I never use_ was causing it, and try uninstalling it (which fixed it, btw).

Thus, Terminal breaks Windows in a fairly ridiculous way, that makes it impossible to track down that it's the cause -- it makes a pretty serious case for "just don't use Terminal"! And I'm extremely happy to take that solution, since I didn't even use the software anyway!

**Comment by FalconFour** [comment]
I didn't even know there was a "open Terminal here" shell extension. I don't even use Windows Terminal - I just had it installed passively -- my habit is still to press the Windows key, type the letters "cmd", and hit Enter or Ctrl+Shift+Enter. Which, on random computers, for months now, had occasionally resulted in cmd (or any other application I open from the start menu) opening up, behind the start menu, leaving me to additionally close the start menu -- or worse, start interacting with the start menu with the keyboard instead of the application (I work very quickly sometimes - this usually happens in the span of 1 second or so).

So, Terminal was identified as the reason for this, eh? And the Terminal folks are like, "😊 haha well it's a Windows bug, we can't fix it"?

I had to become frustrated enough with this issue to even find a way to phrase it to search it - and then trust the absurd conclusion that an app I never use was causing it, and try uninstalling it (which fixed it, btw).

Thus, Terminal breaks Windows in a fairly ridiculous way, that makes it impossible to track down that it's the cause -- it makes a pretty serious case for "just don't use Terminal"! And I'm extremely happy to take that solution, since I didn't even use the software anyway!

What you **DO** have control over is not installing this "open terminal here" extension unless the user asks for it (I would absolutely **not** have wanted it), especially given the nature of this bug -- having been open for well over a year now, unmoved.

**Comment by michaelbunney** [comment]
So the fix for the Windows 10 Start menu bug is to switch to Windows 11, which has stripped the Start menu of its most useful features?  Guess I'll just live without Windows Terminal, because I can't live with Windows 11's version of the Start menu.

**Comment by WagnerGFX** [comment]
> > > Microsoft will continue to support at least one Windows 10 release until October 14, 2025.
> > 
> > 
> > ... plus the usual years-long delays.
> > Is this really proper to just ditch Windows 10 bugs as "won't fix" at the current time? Isn't it too early for that?
> > This saddens me, this bug is really annoying, and it's not even something that only happens inside Windows Terminal, it affects the general OS experience. I am enjoying Terminal but this is making me seriously consider uninstalling it. Please reconsider.
> 
> @pgorod Either of these fixes should work unless you really need that context menu: [#7008 (comment)](https://github.com/microsoft/terminal/issues/7008#issuecomment-662621638) [#7122 (comment)](https://github.com/microsoft/terminal/issues/7122#issuecomment-683299285)

For those who really need the context menu (like me), I created a custom version in the registry that should do the same work, but without breaking the Start menu.

Just be mindful that this solution:
1) Uses the Windows Powershell icon.
2) Requires the "wt" alias, that only exist with the default Terminal installation.
3) Will open in new tabs, whenever possible.

Customize as needed. [Windows Terminal CLI arguments](https://learn.microsoft.com/en-us/windows/terminal/command-line-arguments?tabs=windows)


`Windows Terminal - Win10 Context Menu - Custom (Add).reg`
```
Windows` Registry Editor Version 5.00

[HKEY_CLASSES_ROOT\Directory\Background\shell\WindowsTerminal]
@="Open Terminal here"
"Icon"="C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe"

[HKEY_CLASSES_ROOT\Directory\Background\shell\WindowsTerminal\command]
@="wt -w 0 nt -d \"%v\""
```
